sensor/common/compliance/nodescan_handler_test.go

```diff
@@ -66,70 +65,66 @@ func (s *NodeScanHandlerTestSuite) TearDownTest() {
 	s.cancel()
 }
 
-func (s *NodeScanHandlerTestSuite) startForwardingToCentral(c <-chan *central.MsgFromSensor) {
-	go func() {
-		defer close(s.toCentral)
-		for {
-			select {
-			case <-s.ctx.Done():
-				return
-			case msg, ok := <-c:
-				if !ok {
-					return
-				}
-				s.toCentral <- msg
-			}
-		}
-	}()
-}
-
 // TestStopHandler goal is to stop handler while there are still some messages to process
 // in the channel passed into NewNodeScanHandler.
 // We expect that premature stop of the handler produces no race condition or goroutine leak.
 // Exec with: go test -race -count=1 -v -run ^TestNodeScanHandler$ ./sensor/common/compliance
 func (s *NodeScanHandlerTestSuite) TestStopHandler() {
 	nodeScans := make(chan *storage.NodeScanV2)
 	h := NewNodeScanHandler(nodeScans)
-	// consume all messages send toCentral
+	// simulate Central: consume all messages from h.ResponsesC()
 	go func() {
+		defer close(s.toCentral)
 		for {
 			select {
 			case <-s.ctx.Done():
 				return
-			case <-s.toCentral:
+			case _, ok := <-h.ResponsesC():
+				if !ok {
+					return
+				}
```

Comment:
Simplification: Instead of forwarding to central, we now receive and discard all msgs that are meant to go to central.